lib/rpmrc.cc

```diff
@@ -358,6 +361,105 @@ const char * lookupInDefaultTable(const char * name,
     return name;
 }
 
+static int moveConfigFiles(std::string userdir)
+{
+    fs::path home = std::getenv("HOME");
+    std::error_code ec;
+    std::error_code ec2;
+    fs::path oldmacros = home / ".rpmmacros";
+    fs::path oldrpmrc = home / ".rpmrc";
+    fs::path lockfile = home / ".rpmrc_conversion_lock";
+    fs::path macros_target = "";
+    fs::path rpmrc_target = "";
+
+    if (userdir.substr(0, 2) == "~/")
+	userdir.replace(0, 1, home);
+
+    fs::path userdir_path = userdir;
+    fs::path newmacros = userdir_path / "macros";
+    fs::path newrpmrc = userdir_path / "rpmrc";
+
+    if (!isatty(0)) /* only on interactive runs */
+	return -1;
+
+    if (!fs::is_regular_file(oldmacros) && !fs::is_regular_file(oldrpmrc))
+	return -1;
+
+    int fd = open(lockfile.c_str(), O_CREAT|O_EXCL|O_WRONLY, 0644);
+    if (fd < 0)
+	return -1;
+    close(fd);
+
+    /* recheck after acquiring lock */
+    if ((!fs::is_regular_file(oldmacros) && !fs::is_regular_file(oldrpmrc)) ||
+	fs::exists(userdir_path)) {
+	fs::remove(lockfile, ec);
+	return -1;
+    }
+
+    fs::create_directories(userdir_path, ec);
+    if (ec) goto err2;
+
+    if (fs::is_regular_file(oldmacros)) {
+	fs::copy(oldmacros, newmacros, ec);
```

Comment:
I do dislike the copy. It'll mess timestamps and special permissions for no real reason, and somebody besides me is going to be unhappy about that. I think we should just move these files - if it crosses a filesystem then that's yet another special case we want to leave alone anyhow.

I thought I mumbled something about this earlier but apparently not, guess I deemed it safer at least as of that version, and didn't.